dbt_semantic_interfaces/parsing/where_filter/parameter_set_factory.py

```diff
@@ -108,6 +108,8 @@ def create_entity(entity_name: str, entity_path: Sequence[str] = ()) -> EntityCa
     @staticmethod
     def create_metric(metric_name: str, group_by: Sequence[str] = ()) -> MetricCallParameterSet:
         """Gets called by Jinja when rendering {{ Metric(...) }}."""
+        if not group_by:
+            raise ParseWhereFilterException("`group_by` parameter is required for Metric in where filter.")
```

Comment:
Can we make this a little more descriptive as to why? Maybe something like:

```
"`group_by` parameter is required for filtering on Metrics, since a filter on a Metric does not make sense for queries where there are no sub-groups against which to apply the filter expression."
```

I don't know if normal people would understand that, but maybe it's better?

Reply:
Updated!